File: RuntimeData/xnaruntime/RuntimeXNA/RuntimeXNA/Params/PARAM_EXTENSION.cs
//----------------------------------------------------------------------------------
//
// PARAM_EXTENSION : un parametre extension
//
//----------------------------------------------------------------------------------
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using RuntimeXNA.Application;
using RuntimeXNA.Services;

namespace RuntimeXNA.Params
{	
	public class PARAM_EXTENSION:CParam
	{
		public byte[] data = null;
		
		public override void  load(CRunApp app)
		{
			short size = app.file.readAShort();
			app.file.skipBytes(4); // type + code
			if (size > 6)
			{
				data = new byte[size - 6];
				app.file.read(data);
			}
		}
	}
}

File: RuntimeData/runtime/app/src/main/java/Params/PARAM_EXTENSION.java
/* Copyright (c) 1996-2013 Clickteam
 *
 * This source code is part of the Android exporter for Clickteam Multimedia Fusion 2.
 * 
 * Permission is hereby granted to any person obtaining a legal copy 
 * of Clickteam Multimedia Fusion 2 to use or modify this source code for 
 * debugging, optimizing, or customizing applications created with 
 * Clickteam Multimedia Fusion 2.  Any other use of this source code is prohibited.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 * IN THE SOFTWARE.
 */
//----------------------------------------------------------------------------------
//
// PARAM_EXTENSION : un parametre extension
//
//----------------------------------------------------------------------------------
package Params;

import Application.CRunApp;

public class PARAM_EXTENSION extends CParam
{
    public byte data[]=null;
    
    @Override
	public void load(CRunApp app) 
    {
        short size=app.file.readAShort();
        app.file.skipBytes(4);	    // type + code
	if (size>6)
	{
	    data=new byte[size-6];
	    app.file.read(data);
	}
    }    
    
}
